MapboxCoreNavigation/RouteController.swift

```diff
@@ -371,7 +371,8 @@ extension RouteController: CLLocationManagerDelegate {
             // Don't alert if the route segment is shorter than X
             // However, if it's the beginning of the route
             // There needs to be an alert
-            routeProgress.currentLegProgress.currentStep.distance > minimumDistanceForMediumAlert {
+            routeProgress.currentLegProgress.currentStep.distance > minimumDistanceForMediumAlert &&
+            alertLevel != .high {
```

Comment:
There are other cases where we’re setting the alert level, and if we were to address those cases equally, we’d quickly wind up with spaghetti. Fortunately, `incrementRouteProgressAlertLevel(_:location:)` offers us a handy bottleneck to accomplish the stated intent of this PR: simply bail if the new alert level is “backwards from” (i.e., greater than) the current alert level. It would be necessary to account for the special cases `none`, `depart`, and `arrive`, but overall I think that approach would be more self-documenting and less prone to edge cases.